packages/identity/src/identity/secp256k1.ts

```diff
@@ -0,0 +1,232 @@
+    );
+  }
+
+  public static fromJSON(json: string): Secp256k1KeyIdentity {
+    const parsed = JSON.parse(json);
+    if (Array.isArray(parsed)) {
+      if (typeof parsed[0] === 'string' && typeof parsed[1] === 'string') {
+        return this.fromParsedJson([parsed[0], parsed[1]]);
+      }
+      throw new Error('Deserialization error: JSON must have at least 2 items.');
+    } else if (typeof parsed === 'object' && parsed !== null) {
+      const { publicKey, _publicKey, secretKey, _privateKey } = parsed;
+      const pk = publicKey
+        ? Secp256k1PublicKey.fromRaw(blobFromUint8Array(new Uint8Array(publicKey.data)))
+        : Secp256k1PublicKey.fromDer(blobFromUint8Array(new Uint8Array(_publicKey.data)));
+
+      if (publicKey && secretKey && secretKey.data) {
+        return new Secp256k1KeyIdentity(pk, blobFromUint8Array(new Uint8Array(secretKey.data)));
+      }
+      if (_publicKey && _privateKey && _privateKey.data) {
+        return new Secp256k1KeyIdentity(pk, blobFromUint8Array(new Uint8Array(_privateKey.data)));
+      }
+    }
+    throw new Error(`Deserialization error: Invalid JSON type for string: ${JSON.stringify(json)}`);
+  }
+
+  public static fromKeyPair(publicKey: BinaryBlob, privateKey: BinaryBlob): Secp256k1KeyIdentity {
+    return new Secp256k1KeyIdentity(Secp256k1PublicKey.fromRaw(publicKey), privateKey);
+  }
+
+  public static fromSecretKey(secretKey: ArrayBuffer): Secp256k1KeyIdentity {
+    const publicKey = Secp256k1.publicKeyCreate(new Uint8Array(secretKey), false);
+    const identity = Secp256k1KeyIdentity.fromKeyPair(
+      blobFromUint8Array(publicKey),
+      blobFromUint8Array(new Uint8Array(secretKey)),
+    );
+    return identity;
+  }
+
+  protected _publicKey: Secp256k1PublicKey;
+
+  // `fromRaw` and `fromDer` should be used for instantiation, not this constructor.
+  protected constructor(publicKey: PublicKey, protected _privateKey: BinaryBlob) {
+    super();
+    this._publicKey = Secp256k1PublicKey.from(publicKey);
+  }
+
+  /**
+   * Serialize this key to JSON.
+   */
+  public toJSON(): JsonableSecp256k1Identity {
+    return [blobToHex(this._publicKey.toRaw()), blobToHex(this._privateKey)];
+  }
+
+  /**
+   * Return a copy of the key pair.
+   */
+  public getKeyPair(): {
+    secretKey: BinaryBlob;
+    publicKey: Secp256k1PublicKey;
+  } {
+    return {
+      secretKey: blobFromUint8Array(new Uint8Array(this._privateKey)),
+      publicKey: this._publicKey,
+    };
+  }
+
+  /**
+   * Return the public key.
+   */
+  public getPublicKey(): PublicKey {
+    return this._publicKey;
+  }
+
+  /**
+   *  Return private key in a pem file
+   */
+
+  public getPem(): string {
+    const rawPrivateKey = this._privateKey.toString('hex');
+    const rawPublicKey = this._publicKey.toRaw().toString('hex');
+
+    return `${PEM_BEGIN}\n${Buffer.from(
+      `${PRIV_KEY_INIT}${rawPrivateKey}${KEY_SEPARATOR}${rawPublicKey}`,
+      'hex',
+    ).toString('base64')}\n${PEM_END}`;
+  }
```

Comment:
@robin-kunzler we added this feature to export the identity in PEM format, to be imported into dfx. 
Let me know if you want this removed to keep the functionality consistent with the ed25519 identity.

Reply:
I think I would remove the PEM export, since we expect this to run in browsers, and that could potentially lead to some security issues